opal/models.py

```diff
@@ -1684,6 +1701,35 @@ def explicit_access_only(self):
         return any(r for r in all_roles if r == "scientist")
 
 
+class Stage(TrackedModel):
+    """ An episode can have many stages, but should have no more than
+        one with no stop date any time.
+
+        This is expected to be set by the episode_category
+        which determines what stages are available to an episode.
+    """
+    episode = models.ForeignKey(Episode, null=False)
+    started = models.DateTimeField()
+    stopped = models.DateTimeField(blank=True, null=True)
+    value = models.CharField(max_length=256)
+
+    @transaction.atomic
+    def save(self, *args, **kwargs):
+        """ An episode should not have multiple open stages.
+            raise exceptions and roll back if anyone tries
+            any funny business.
+        """
+        super(Stage, self).save(*args, **kwargs)
+        existing = Stage.objects.filter(episode=self.episode)
+        existing = existing.filter(stopped=None)
+        if existing.count() > 1:
```

Comment:
You're relying on transaction.atomic rolling it back and not saving ?
Does a unique_together constraint not work?

Reply:
so we can do a unique_together in theory but its a bit of a fudge.

What we want 
```
   unique_together = (('episode', 'value', 'stopped'),)
```
but really we 'd like
```
   unique_together = (('episode', 'value', 'stopped' != None),)
```
Although as they are datetime field they are probably the same thing... I thought this was was a bit nicer as its an explicit check